trade-finance-manager-ui/server/controllers/case/underwriting/amendments/amendment-lead-underwriter/index.js

```diff
@@ -0,0 +1,175 @@
+  const {
+    _id: dealId,
+    amendmentId,
+    facilityId,
+  } = req.params;
+  const deal = await api.getDeal(dealId);
+
+  const { data: amendment } = await api.getAmendmentById(facilityId, amendmentId);
+
+  // if no amendment or deal
+  if ((!amendment || !amendment.amendmentId) || !deal) {
+    return res.redirect('/not-found');
+  }
+
+  const { user } = req.session;
+
+  // checks if can edit
+  const userCanEdit = canUserEditLeadUnderwriter(user);
+
+  if (!userCanEdit) {
+    return res.redirect('/not-found');
+  }
+
+  let currentLeadUnderWriterUserId;
+
+  // if already set then shows name of assigned underwriter
+  if (amendment.leadUnderwriter) {
+    currentLeadUnderWriterUserId = amendment.leadUnderwriter.userId;
+  }
+
+  // gets all underwriter and managers
+  const allUnderwriterManagers = await api.getTeamMembers(CONSTANTS.TEAMS.UNDERWRITER_MANAGERS);
+  const allUnderwriters = await api.getTeamMembers(CONSTANTS.TEAMS.UNDERWRITERS);
+
+  const allTeamMembers = [
+    ...allUnderwriterManagers,
+    ...allUnderwriters,
+  ];
+
+  // sorts alphabetically
+  const alphabeticalTeamMembers = sortArrayOfObjectsAlphabetically(allTeamMembers, 'firstName');
+
+  return res.render('case/underwriting/amendments/amendment-lead-underwriter/amendment-assign-lead-underwriter.njk', {
+    activeSubNavigation: 'underwriting',
+    deal: deal.dealSnapshot,
+    tfm: deal.tfm,
+    dealId: deal.dealSnapshot._id,
+    user,
+    assignToSelectOptions: mapAssignToSelectOptions(currentLeadUnderWriterUserId, user, alphabeticalTeamMembers),
+    amendment,
+  });
+};
+
+/**
+ * @param {*} req
+ * @param {*} res
+ * checks if unassigned or assigned to user
+ * if unassigned then does not make getUser api call
+ * redirects back to underwriting page
+ */
+const postAssignAmendmentLeadUnderwriter = async (req, res) => {
+  const {
+    _id: dealId,
+    amendmentId,
+    facilityId,
+  } = req.params;
+  const deal = await api.getDeal(dealId);
+
+  const { data: amendment } = await api.getAmendmentById(facilityId, amendmentId);
+
+  if ((!amendment || !amendment.amendmentId) || !deal) {
+    return res.redirect('/not-found');
+  }
+
+  const { user } = req.session;
+
+  const userCanEdit = canUserEditLeadUnderwriter(user);
+
+  if (!userCanEdit) {
+    return res.redirect('/not-found');
+  }
+
+  const {
+    assignedTo: assignedToValue, // will be user._id or `Unassigned`
+  } = req.body;
+
+  if (assignedToValue === 'Unassigned') {
+    const update = {
+      leadUnderwriter: {
+        userId: assignedToValue,
+        firstName: '',
+        lastName: '',
+        email: '',
+      },
+    };
+
+    await api.updateAmendment(facilityId, amendmentId, update);
+
+    return res.redirect(`/case/${dealId}/underwriting`);
+  }
+
+  const underWriterUserToAdd = await api.getUser(assignedToValue);
+
+  const update = {
+    leadUnderwriter: {
+      userId: assignedToValue,
+      firstName: underWriterUserToAdd.firstName,
```

Comment:
same as above - only the Id
(i.e. leadUnderwriterId: `1234`)

Reply:
done